File: demos/jans-chip/ios/Jans-Chip/Models/AppIntegrityResponse.swift
//
//  AppIntegrityResponse.swift
//  Jans-Chip
//
//  Created by Nazar Yavornytskyi on 12.10.2023.
//

import Foundation

struct AppIntegrityResponse: Codable {
    
    var requestDetails: RequestDetails
    var appIntegrity: AppIntegrity
    var deviceIntegrity: DeviceIntegrity
    var accountDetails: AccountDetails
    var error: String
    
    var isSuccessful: Bool = false
}

struct RequestDetails: Codable {
    var requestPackageName: String
    var timestampMillis: String
    var nonce: String
}

struct AppIntegrity: Codable {
    var requestPackageName: String
}

struct DeviceIntegrity: Codable {
    var deviceRecognitionVerdict: [String]
}

struct AccountDetails: Codable {
    var appLicensingVerdict: String
}

File: demos/jans-chip/android/app/src/main/java/io/jans/chip/model/appIntegrity/AppIntegrityResponse.kt
package io.jans.chip.model.appIntegrity

import androidx.room.Ignore
import com.google.gson.annotations.SerializedName

data class AppIntegrityResponse(
    @SerializedName("requestDetails")
    var requestDetails: RequestDetails? = null,

    @SerializedName("appIntegrity")
    var appIntegrity: AppIntegrity? = null,

    @SerializedName("deviceIntegrity")
    var deviceIntegrity: DeviceIntegrity? = null,

    @SerializedName("accountDetails")
    var accountDetails: AccountDetails? = null,

    @SerializedName("error")
    var error: String? = null
) {
    @Ignore
    var isSuccessful: Boolean? = false

    @Ignore
    var errorMessage: String? = null
}